packages/flutter_tools/lib/src/base/utils.dart

```diff
@@ -104,6 +104,13 @@ class ItemListNotifier<T> {
     removedItems.forEach(_removedController.add);
   }
 
+  void removeItem(T item) {
+    if (_items.contains(item)) {
```

Comment:
nit, you can simplify this a bit since `List.remove` returns whether it successfully removed the item:

```
if (_items.remove(item)) {
  _removedController.add(item);
}
```